### PR TITLE
Fixed #25163 -- Fixed exception handling in nested test client requests.

### PR DESCRIPTION
**Pull request by pscottdevos**
This solves the problem with the following ticket

https://code.djangoproject.com/ticket/25249#ticket

A unit test is provided.

### CODE REVIEW

**Comment by pscottdevos** [comment]
commit passes all unit tests in master and stable-1.8.x branches.

**Comment by pscottdevos** [comment]
This commit passes all checks (hooray!) (as well as all unit tests)

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: tests/test_client/views.py [hunk 1]
```diff
@@ -309,5 +310,16 @@ def mass_mail_sending_view(request):
...
 
+def nesting_exception_view(request):
+    """A simple view that uses a nested client to call another view and
+    then raises an exception
+    """
+    client = Client()
+    data = {'var': '\xf2'}
+    response = client.get('/get_view/', data)
+    raise Exception('exception message')
+    return response
```
Comment (coldmind) [inline tests/test_client/views.py]: unneeded return -> all things above except `raise` statement are not needed too

Comment (pscottdevos) [inline tests/test_client/views.py]: The bug found is caused by the nested client.get call. If you remove the the stuff above the raise statement, you aren't testing the provided fix.

**Comment by timgraham** [comment]
merged in 7fa1dd8a80795d7573fe650014c5d23af7557bc7, thanks!

**Pull request closed without merging**